### ISSUE TITLE
Framework:  Create build to protect UVM off

### ISSUE THREAD

**Issue by jwillenbring** [trilinos/Trilinos#8397]
## Enhancement
@trilinos/framework 

Once either #7047 or #8129 has been completed, we need to create another PR build that turns UVM off. We need to start with the subset that can be built cleanly with UVM off and add packages as they are cleaned up.

**Comment by csiefer2** [comment]
I suspect you can start with Teuchos :)

**Comment by rppawlo** [comment]
Phalanx is ready if you want to use that as a test case.

**Comment by jwillenbring** [comment]
With the #8482 PR submitted which will satisfy #7047, I want to lay out a proposed plan for completing this issue. We can iterate on this as it makes sense. As we complete the steps in the plan we decide upon, I want to track the completion of each step in this ticket. Here is my proposed plan:

1) Make changes to turn UVM off to a branch and fire off a test build. Paste a link to the results in this ticket.

1) Once we have a build reflective of the overall state, set up a nightly specialized build on the dashboard with all packages (at least those relevant to the build) enabled. Leave this build running until the end of the effort to establish a fully clean launch blocking=off build, including having a PR build that protects that functionality.

1) Establish a new PR build for UVM OFF that enables only those packages working at 100%. As @rppawlo says above, we can start with just Phalanx, but before we can call this done, we need to get all additional packages that are clean at that time enabled. We can add to this build as additional packages are cleaned up. However, when this build is fully in place, with all currently clean packages enabled, this ticket as defined is done.

@trilinos/framework Let's discuss this plan of action.

**Comment by ZUUL42** [comment]
I've setup [a branch](https://github.com/trilinos/Trilinos/compare/develop...ZUUL42:uvm) for the CUDA 10.1.105 build that turns on UVM for Kokkos & Tpetra.
Running it through the Jenkins job, I get this. [CDash](https://testing.sandia.gov/cdash/index.php?project=Trilinos&parentid=8435101)
I haven't disabled any packages for this yet.

**Comment by ZUUL42** [comment]
After turning off 4 packages, @trilinos/panzer , @trilinos/stokhos, @trilinos/intrepid2, & @trilinos/TrilinosCouplings, it now builds without error.
However, we still have 445 failing tests reporting in 17 packages. [CDash](https://testing.sandia.gov/cdash/index.php?project=Trilinos&display=project&filtercount=3&showfilters=1&filtercombine=and&field1=site&compare1=61&value1=ride18&field2=buildname&compare2=61&value2=PR-uvm_4packagesoff-test-Trilinos_pullrequest_cuda_10.1.105-143&field3=buildstamp&compare3=61&value3=20210108-1756-Experimental)

**Comment by ZUUL42** [comment]
After running through a few test builds, I've turned off a significant number of tests that fail in this UVM = OFF build. This is not intended to be its final state. It will allow us to get this build in place as a PR and/or dev->master build and then we can get the tests back in place from there.

I'll have the PR in soon.
Once I do ...

As package teams modify tests to pass, they can be turned back on by removing the associated line in:
[`cmake\std\PullRequestLinuxCuda10.1.105uvmTestingSettings.cmake`](https://github.com/trilinos/Trilinos/blob/develop/cmake/std/PullRequestLinuxCuda10.1.105uvmTestingSettings.cmake)
Teams can do this, just request a review from any of the Framework team and the test will be reenabled for this build.

For progress tracking, to start the UVM = OFF build, I have turned off tests for:
- Tpetra 222
- Zoltan2 133
- MueLu 78
- ShyLU_DD 60
- Ifpack2 44
- Domi 38
- Kokkos 35
- Belos 33
- Phalanx 31
- Sacado 21
- STK 18
- Teko 18
- Xpetra 17
- ROL 16
- KokkosKernels 16
- Anasazi 10
- Amesos2 9
- NOX 8
- SEACAS 6
- Adelus 4
- Thyra 3
- Teuchos 3
- Piro 2
- MiniTensor 2
- Stratimikos 1

**Comment by ZUUL42** [comment]
This build has now been added to the dev->master build set.

**Comment by jwillenbring** [comment]
@mmwolf mentioned yesterday he does not see a PR build as a high priority, or possibly even desirable in the near term. The dev->master build is a good solution for the near term. We should perhaps rescope this to that level, and close it after everything is stablized at that level.

**Comment by ZUUL42** [comment]
We will review the state of this later or if any issues come up.
